### PR TITLE
Format skills into extract skills 

### PR DESCRIPTION
**Pull request by india-kerle**
I ended up adding a parameter in extract_skills method to format a skills list instead to format skills - very minor changes! I've also added an example config file.

This PR would close:
- #101 
- #99

### CODE REVIEW

File: ojd_daps_skills/config/extract_skills_template.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#This is a template config file - we have added definitions to parameters that you will need to modify for your own taxonomy
+
+#the relative path to the trained NER model
+ner_model_path: "outputs/models/ner_model/20220825/"
+#the relative path to where
+taxonomy_path: "path/to/formatted_taxonomy.csv"
+#the name of your own taxonomy
+taxonomy_name: "your_taxonomy_name"
```
Comment (india-kerle) [inline ojd_daps_skills/config/extract_skills_template.yaml:8]: keeping this for now but we can get rid of it if we end up just using the config name a la #106

File: ojd_daps_skills/config/extract_skills_template.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#This is a template config file - we have added definitions to parameters that you will need to modify for your own taxonomy
```
Comment (lizgzil) [inline ojd_daps_skills/config/extract_skills_template.yaml:1]: nice!

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py [hunk 3]
```diff
@@ -365,22 +384,30 @@ def map_skills(self, predicted_skills):
 
         return job_skills_matched_formatted
 
-    def extract_skills(self, job_adverts, map_to_tax=True):
+    def extract_skills(
+        self, job_adverts_skills: Union[str, List[dict]], format_skills=False
```
Comment (lizgzil) [inline ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: should this be `Union[str, List[str]]`  - a single job advert, a list of job adverts or a list of skills?

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py [hunk 4]
```diff
@@ -365,22 +384,30 @@ def map_skills(self, predicted_skills):
...
+    ):
         """
-        Extract skills using the NER model from a single or a list of job adverts
-        and if map_to_tax==True then also map them to the taxonomy
+        Extract skills using the NER model from:
+            1) A single job advert;
+            2) A list of job adverts or;
+            3) A list of skills (can contain multiskills) 
+        if format==True then format skills list and also map them to the taxonomy
```
Comment (lizgzil) [inline ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: I think this needs to be clearer:
```
if you are entering a list of skills, then you need to set `format=True` in order to map them to a taxonomy
```

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py [hunk 5]
```diff
@@ -259,14 +277,15 @@ def get_skills(self, job_adverts):
                 else:
                     skills[label].append(ent_text)
             predicted_skills.append(skills)
+
         return predicted_skills
 
-    def map_skills(self, predicted_skills):
+    def map_skills(self, predicted_skills:List[str]):
```
Comment (lizgzil) [inline ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: Thinking about it, I think we might want to include the functionality here too. If I had a list of skills I'd expect to be able to use this function to map them. It feels like only a minor edition too

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py (hunk 3 above)
Comment (lizgzil) [inline ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: I feel funny about this. It does feel odd that you would use a function called "extract_skills" on a list of skills..

**Review by lizgzil — COMMENTED** [review]
on second thoughts - I feel weird this functionality isn't available in `map_skills`. It feels like it should at least be in both? Unless I'm missing something?

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py (hunk 3 above)
Comment (india-kerle) [reply ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: oh ya - good point !

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py (hunk 5 above)
Comment (india-kerle) [reply ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: ok gotcha - I'll add it to both then

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py (hunk 3 above)
Comment (india-kerle) [reply ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: deal

**Comment by india-kerle** [comment]
@lizgzil i've added functionality to map_skills, some type hints and fleshed out some doc strings. let me know if you think its ok to merge.

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py [hunk 6]
```diff
@@ -109,10 +111,10 @@ def __init__(self, config_name="extract_skills_toy", local=True, verbose=True):
 
     def load(
         self,
-        taxonomy_embedding_file_name=None,
-        prev_skill_matches_file_name=None,
-        hard_labelled_skills_name=None,
-        hier_name_mapper_file_name=None,
+        taxonomy_embedding_file_name:Optional[str]=None,
```
Comment (lizgzil) [inline ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: cool

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py [hunk 7]
```diff
@@ -280,10 +280,16 @@ def get_skills(self, job_adverts):
...
 
-    def map_skills(self, predicted_skills:List[str]):
+    def map_skills(self, predicted_skills: Union[List[dict], List[str]]):
         """
         Maps a list of skills to a skills taxonomy
+
+        If predicted_skills is a list of skills, format it accordingly to
+            be mapped to a skills taxonomy. 
         """
+        if all(isinstance(predicted_skill, str) for predicted_skill in predicted_skills):
```
Comment (lizgzil) [inline ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: I think you could check the first one to make this speedier? `if isinstance(predicted_skills[0], str)`?

Comment (lizgzil) [reply ojd_daps_skills/pipeline/extract_skills/extract_skills.py]:
```suggestion
        if isinstance(predicted_skills[0], str):
```

**Review by lizgzil — APPROVED** [review]
looks good!

File: ojd_daps_skills/pipeline/extract_skills/extract_skills.py (hunk 7 above)
Comment (india-kerle) [reply ojd_daps_skills/pipeline/extract_skills/extract_skills.py]: ya that would work too - but thinking what if a user puts in a list with things other than strings? its a bit of an edge case i'd guess !

**Pull request merged**